dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -691,27 +681,26 @@ public String getHTML(String uri, Host host, boolean liveMode, String contentId,
             ).request();
         HttpServletResponse responseProxy = new BaseResponse().response();
 
-        StringWriter out = new StringWriter();
-        Context context = null;
-
 
+        Identifier ident = identifierAPI.find(host, uri);
+        if (ident==null || !UtilMethods.isSet(ident.getId()) ) {
+            throw new ResourceNotFoundException(String.format("Resource %s not found in Live mode!", uri));
+        }
 
-        // Map with all identifier inodes for a given uri.
-        String idInode = identifierAPI.find(host, uri).getId();
         
 
-        ContentletVersionInfo cinfo = APILocator.getVersionableAPI().getContentletVersionInfo( idInode, langId );
+        ContentletVersionInfo cinfo = APILocator.getVersionableAPI().getContentletVersionInfo( ident.getId(), langId );
         if(cinfo==null && langId!=APILocator.getLanguageAPI().getDefaultLanguage().getId()){
-          cinfo = APILocator.getVersionableAPI().getContentletVersionInfo( idInode, APILocator.getLanguageAPI().getDefaultLanguage().getId() );
+          cinfo = APILocator.getVersionableAPI().getContentletVersionInfo( ident.getId(), APILocator.getLanguageAPI().getDefaultLanguage().getId() );
         }
         // if we still have nothing.
-        if (!InodeUtils.isSet(idInode) || cinfo.getLiveInode() == null) {
+        if (!InodeUtils.isSet(ident.getId()) || cinfo.getLiveInode() == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "cinfo" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)